src/concrete_solve.jl

```diff
@@ -372,11 +372,6 @@ function DiffEqBase._concrete_solve_adjoint(
         saveat = eltype(prob.tspan)[],
         save_idxs = nothing,
         kwargs...)
-    if !(sensealg isa GaussAdjoint) &&
```

Comment:
If lifting this, then InterpolatingAdjoint and QuadartureAdjoint should also be tested.

Reply:
Agreed, that's why I have marked this draft, so I can test, and if necessary add the check back with allowing BacksolveAdjoint through 😅